src/Plugin/Field/FieldWidget/Select2Widget.php

```diff
@@ -23,13 +23,58 @@
  */
 class Select2Widget extends OptionsSelectWidget {
 
+  /**
+   * {@inheritdoc}
+   */
+  public static function defaultSettings() {
+    return [
+      'autocomplete' => FALSE,
+    ] + parent::defaultSettings();
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function settingsForm(array $form, FormStateInterface $form_state) {
+    $element = [];
+    if ($this->fieldDefinition->getType() == 'entity_reference') {
```

Comment:
Not sure about the field type check here. Since it does not scale well with other field types like Entity Reference Revisions. We need multiple widgets anyway at some point.